iOS/DuckDuckGo/DownloadManager.swift

```diff
@@ -105,7 +105,10 @@ class DownloadManager: DownloadManaging {
     }
 
     func downloadMetaData(for response: URLResponse, suggestedFilename: String? = nil) -> DownloadMetadata? {
-        let filename = filename(forSuggestedFilename: suggestedFilename ?? response.suggestedFilename, mimeType: response.mimeType)
+        guard let filename = try? filename(forSuggestedFilename: suggestedFilename ?? response.suggestedFilename,
```

Comment:
I understand that this is enlarging the scope of the fix, and you may decide to postpone this change, but using nil as a response when we have a failure it's a risky approach.
I would suggest making this function throw, so the return value is clear:
- A value: all ok
- Nil: the file doesn't exist
- throw an error: something went wrong

Reply:
This discussion is related: https://github.com/duckduckgo/apple-browsers/pull/2880#discussion_r2610993390
@dus7 thoughts?